Sources/Kitura/Kitura.swift

```diff
@@ -64,13 +64,48 @@ public class Kitura {
                                     withSSL sslConfig: SSLConfig?=nil,
                                     keepAlive keepAliveState: KeepAliveState = .unlimited,
                                     allowPortReuse: Bool = false) -> HTTPServer {
+        return Kitura._addHTTPServer(onPort: port, with: delegate, withSSL: sslConfig, keepAlive: keepAliveState, allowPortReuse: allowPortReuse)
+    }
+
+    /// Add an HTTPServer on a Unix domain socket path with a delegate.
+    ///
+    /// The server is only registered with the framework, it does not start listening
+    /// on the Unix socket until `Kitura.run()` or `Kitura.start()` are called.
+    ///
+    ///### Usage Example: ###
+    ///```swift
+    /// let router = Router()
+    /// Kitura.addHTTPServer(onUnixDomainSocket: "/tmp/mySocket", with: router)
+    ///```
+    /// - Parameter onUnixDomainSocket: The path of the Unix domain socket to listen on.
+    /// - Parameter with: The `ServerDelegate` to use.
+    /// - Parameter withSSL: The `sslConfig` to use.
+    /// - Parameter keepAlive: The maximum number of additional requests to permit per Keep-Alive connection. Defaults to `.unlimited`. If set to `.disabled`, Keep-Alive will not be permitted.
+    /// - Returns: The created `HTTPServer`.
+    @discardableResult
+    public class func addHTTPServer(onUnixDomainSocket socketPath: String,
+                                    with delegate: ServerDelegate,
+                                    withSSL sslConfig: SSLConfig?=nil,
+                                    keepAlive keepAliveState: KeepAliveState = .unlimited) -> HTTPServer {
+        return Kitura._addHTTPServer(onUnixDomainSocket: socketPath, with: delegate, withSSL: sslConfig, keepAlive: keepAliveState)
+    }
+
+    private class func _addHTTPServer(onPort port: Int? = nil, onUnixDomainSocket socketPath: String? = nil,
+                                    with delegate: ServerDelegate,
+                                    withSSL sslConfig: SSLConfig?=nil,
+                                    keepAlive keepAliveState: KeepAliveState = .unlimited,
+                                    allowPortReuse: Bool = false) -> HTTPServer {
         let server = HTTP.createServer()
         server.delegate = delegate
         server.sslConfig = sslConfig?.config
         server.keepAliveState = keepAliveState
         server.allowPortReuse = allowPortReuse
         serverLock.lock()
-        httpServersAndPorts.append((server: server, port: port))
+        if let port = port {
+            httpServersAndPorts.append((server: server, port: port))
+        } else if let socketPath = socketPath {
+            httpServersAndUnixSocketPaths.append((server: server, socketPath: socketPath))
+        }
```

Comment:
What happens if both `port` and `socketPath` are nil? Silent failure. I know this is impossible today but let's add a final `else{}` here for safety.

Reply:
I changed this to use a private enum: https://github.com/IBM-Swift/Kitura/pull/1434/commits/026f967f56e4300f21cf5e740dd59c3a5d1e4615